test/functional/vimscript/system_spec.lua

```diff
@@ -660,4 +660,16 @@ describe('shell :!', function()
         1]])
     end
   end)
+
+  it(':{range}! with powershell without redirecting to buffer', function()
+    local screen = Screen.new(500, 10)
+    screen:attach()
+    helpers.set_shell_powershell(true)
+    insert([[ NeoVim is awesome! ]])
+    feed(':4verbose %w !curl --silent --show-error http://nonexisting1234.com<cr>')
```

Comment:
can we avoid curl? making a bogus network request in tests could have weird failures. Would it work to use nvim itself (`v:progpath`)?
```suggestion
    feed('exe ":4verbose %w !" v:progpath "--clean"<cr>')
```

Reply:
Yes.  I replaced it with a call to `sort` though.  Seemed easier.  Let's see if the tests pass this time.

Reply:
As can be seen from #20890, it is quite useful to be able to test powershell behaviour from non-windows machines. I don't have Windows.

Reply:
> it is quite useful to be able to test powershell behaviour from non-windows machines

Makes sense.  One thing to consider, though, is that the unix version of pwsh has several weird behaviors.  I faced two myself.  One I forgot, but one can be reproduced by these steps:

0. `pwsh`
1. `printf '2\n1' > unsorted.txt`
2. `Start-Process sort -RedirectStandardInput ./unsorted.txt -RedirectStandardOutput ./sorted.txt`
3. `cat sorted.txt`

You'll see that the newline is vanished in `sorted.txt` file.  But if you use `sort` command directly, without invoking it through `Start-Process`, you don't have this problem.

I asked about this, and the other oddity I faced, on PowerShell's discord server.  Nobody gave a legitimate reason for it.  Few even cared, because it was a behavior in non-Windows machines.

More to the point of this PR, though, is that for some reason (I don't have deep understanding of how the tests are carried out, so I'll have to resort to a nebulous "some"), my test is getting stuck _after passing_, in unix machines.  It could be one of those non-Windows oddities; it could be a problem in the test suite; it could any one of a hundred things.  My reason for not wanting to test for pwsh in unix machines is simply to relieve myself from hunting down the problem, which frankly won't even be faced by the user, because 1) no unix user is using powershell and 2) this test is passing, and nvim works perfectly when I perform the steps of the test manually.

Reply:
> As can be seen from https://github.com/neovim/neovim/pull/20890, it is quite useful to be able to test powershell behaviour from non-windows machines.

Evidently not, since the behavior isn't the same.